File: utils/src/main/java/io/github/aemogie/timble/utils/StringUtils.java
package io.github.aemogie.timble.utils;

import java.util.regex.Pattern;

public final class StringUtils {
	private StringUtils() {}
	
	public static final Pattern WORD_REGEX = Pattern.compile("\\w+");
	
	public static String integerWithSpecifiedDigits(int val, int noOfDigits) {
		String valStr = String.valueOf(val);
		if (valStr.length() < noOfDigits) {
			valStr = "0".repeat(noOfDigits - valStr.length()) + valStr;
		}
		return valStr;
	}
}

File: utils/src/main/kotlin/io/github/aemogie/timble/utils/StringUtils.kt
package io.github.aemogie.timble.utils

import java.util.regex.Pattern

object StringUtils {
	val WORD_REGEX = Pattern.compile("\\w+")
	fun integerWithSpecifiedDigits(`val`: Int, noOfDigits: Int): String {
		var valStr = `val`.toString()
		if (valStr.length < noOfDigits) {
			valStr = "0".repeat(noOfDigits - valStr.length) + valStr
		}	
		return valStr
	}
}